metadata-ingestion/scripts/capability_summary.py

```diff
@@ -0,0 +1,231 @@
+
+from datahub.ingestion.api.decorators import SupportStatus
+from datahub.ingestion.source.source_registry import source_registry
+
+logger = logging.getLogger(__name__)
+
+
+def load_plugin_capabilities(plugin_name: str) -> Optional[Plugin]:
+    """Load plugin capabilities without generating full documentation."""
+    logger.debug(f"Loading capabilities for {plugin_name}")
+
+    try:
+        class_or_exception = source_registry._ensure_not_lazy(plugin_name)
+        if isinstance(class_or_exception, Exception):
+            # Log the specific error but don't re-raise it
+            logger.warning(f"Plugin {plugin_name} failed to load: {class_or_exception}")
+            return None
+        source_type = source_registry.get(plugin_name)
+        logger.debug(f"Source class is {source_type}")
+
+        if hasattr(source_type, "get_platform_name"):
+            platform_name = source_type.get_platform_name()
+        else:
+            platform_name = plugin_name.title()
+
+        platform_id = None
+        if hasattr(source_type, "get_platform_id"):
+            platform_id = source_type.get_platform_id()
+        if platform_id is None:
+            logger.warning(f"Platform ID not found for {plugin_name}")
+            return None
+
+        plugin = Plugin(
+            name=plugin_name,
+            platform_id=platform_id,
+            platform_name=platform_name,
+            classname=".".join([source_type.__module__, source_type.__name__]),
+        )
+
+        if hasattr(source_type, "get_support_status"):
+            plugin.support_status = source_type.get_support_status()
+
+        if hasattr(source_type, "get_capabilities"):
+            capabilities = list(source_type.get_capabilities())
+            if capabilities:
+                capabilities.sort(key=lambda x: x.capability.value)
+                plugin.capabilities = capabilities
+            else:
+                logger.debug(f"No capabilities defined for {plugin_name}")
+                plugin.capabilities = []
+        else:
+            logger.debug(f"No get_capabilities method for {plugin_name}")
+            plugin.capabilities = []
+
+        return plugin
+
+    except Exception as e:
+        logger.warning(f"Failed to load capabilities for {plugin_name}: {e}")
+        return None
+
+
+@dataclasses.dataclass
+class CapabilitySummary:
+    """Summary of capabilities across all plugins."""
+
+    plugin_details: Dict[str, Dict]  # plugin_name -> detailed info
+
+
+def generate_capability_summary() -> CapabilitySummary:
+    """Generate a comprehensive summary of capabilities across all plugins."""
+
+    plugin_details: Dict[str, Dict] = {}
+
+    for plugin_name in sorted(source_registry.mapping.keys()):
+        if plugin_name in {
+            "snowflake-summary",
+            "snowflake-queries",
+            "bigquery-queries",
+        }:
+            logger.info(f"Skipping {plugin_name} as it is on the deny list")
+            continue
+
```

Comment:
should we make deny-list a constant at the top?

Reply:
will do